### PR TITLE
Fix Todo.duration and Event.duration to return DURATION property when set (fixes #867)

### LINKED ISSUE #867: [BUG] todo.duration raises error if dtstart is missing
## Describe the bug

While an event should have a dtstart, there is no such requirement for a task - and it may make perfect sense to give a task a duration (a time estimate) without giving a dtstart.  However, `task.duration` currently yields an error

## To Reproduce

```
from icalendar import Todo
from datetime import timedelta

my_task = Todo.from_ical("""BEGIN:VTODO
UID:taskwithoutdtstart
DTSTAMP:20070313T123432Z
DURATION:P5D
SUMMARY:This is a task that is expected to take five days to complete
STATUS:NEEDS-ACTION
END:VTODO""")

assert(my_task.duration == timedelta(days=5))
```

Output:
```
Traceback (most recent call last):
  File "/tmp/foobar.py", line 12, in <module>
    assert(my_event.duration == timedelta(days=5))
           ^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.13/site-packages/icalendar/cal.py", line 1077, in duration
    return self.end - self.start
           ^^^^^^^^
  File "/usr/lib/python3.13/site-packages/icalendar/cal.py", line 1062, in end
    raise IncompleteComponent("No DUE or DURATION+DTSTART given.")
icalendar.error.IncompleteComponent: No DUE or DURATION+DTSTART given....
```

## Expected behavior

`my_task.duration` should return 5 days.

## Additional context
commit d543e2c in the caldav project currently breaks

```
$ pytest -k testTodoDuration
(...)
FAILED test_caldav_unit.py::TestCalDAV::testTodoDuration - icalendar.error.IncompleteComponent: No DUE or DURATION+DTSTART given.
```

### PR DESCRIPTION
**Pull request by SashankBhamidi**
## Summary

This PR fixes issue #867 where `Todo.duration` raises an `IncompleteComponent` error when the `DURATION` property is set but `DTSTART` is missing. The fix ensures that both `Todo.duration` and `Event.duration` return the explicit `DURATION` property value when it exists, before falling back to calculated duration from start/end times.

### Changes Made

- **Fixed `Todo.duration`** (src/icalendar/cal/todo.py:215-226): Now checks for explicit `DURATION` property first, then falls back to calculated duration
- **Fixed `Event.duration`** (src/icalendar/cal/event.py:305-316): Applied same fix for consistency across components  
- **Added comprehensive tests** (src/icalendar/tests/test_issue_867_todo_duration_fix.py): 12 test cases covering all scenarios including the exact reproduction from issue #867
- **Updated existing tests** (src/icalendar/tests/test_issue_662_component_properties.py:393-423): Separated components with DURATION property from incomplete components and added new test validating the fix
- **Updated documentation** (CHANGES.rst): Added entry documenting the bug fix

### Problem Solved

Before this fix, `Todo.duration` would always try to calculate duration from `DTSTART` and `DUE/DTEND`, causing an `IncompleteComponent` error when `DTSTART` was missing - even if an explicit `DURATION` property was set. This violated RFC 5545 which allows `DURATION` to exist without `DTSTART` for todos.

The caldav project was affected by this bug, as their tests expect `Todo.duration` to work with explicit `DURATION` properties.

### RFC 5545 Compliance

This fix ensures proper RFC 5545 compliance:
- For todos: `DURATION` can exist without `DTSTART` (as per RFC 5545)
- For events: `DURATION` property takes precedence over calculated duration
- Maintains 100% backward compatibility for existing functionality

### Test Coverage

- ✅ Exact reproduction of issue #867 scenario
- ✅ Backward compatibility with existing DTSTART+DUE/DTEND usage
- ✅ Property precedence (explicit DURATION vs calculated)
- ✅ Complex duration formats (ISO 8601)
- ✅ Edge cases and error conditions
- ✅ Both Todo and Event components covered

All existing tests pass, confirming no regressions.

Fixes #867

----
📚 Documentation preview 📚: https://icalendar--880.org.readthedocs.build/

### CODE REVIEW

**Comment by stevepiercy** [comment]
Hi @SashankBhamidi, I'm so sorry, evidently the `tox -e ruff` command applied reformatting on all files, but we should have done only the files you touched in the original PR. My mistake.

We should do reformatting of the entire codebase in a separate PR to keep your changes clean. Would you please revert the changes in the other files that have nothing to do with your original PR?

Again, my apologies for the bad instructions. We definitely need to document this procedure.

**Comment by SashankBhamidi** [comment]
> Hi @SashankBhamidi, I'm so sorry, evidently the `tox -e ruff` command applied reformatting on all files, but we should have done only the files you touched in the original PR. My mistake.
> 
> We should do reformatting of the entire codebase in a separate PR to keep your changes clean. Would you please revert the changes in the other files that have nothing to do with your original PR?
> 
> Again, my apologies for the bad instructions. We definitely need to document this procedure.

Thanks for the clarification! I've reverted the excessive ruff formatting and now only apply formatting to the files specifically modified for this feature.

**Comment by niccokunzmann** [comment]
> Thanks for the clarification! I've reverted the excessive ruff formatting and now only apply formatting to the files specifically modified for this feature.

If you install the pre-commit, then this would be done  on every commit automatically.

File: src/icalendar/tests/test_issue_867_todo_duration_fix.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    assert todo["DURATION"].dt == timedelta(hours=2)
+
+    # And our computed property should match
+    assert todo.duration == timedelta(hours=2)
+
+
+# Event tests - same fix applies for consistency
+def test_event_duration_prefers_duration_property():
+    """Test that Event.duration also prefers `DURATION` property over calculated duration."""
+    event = Event()
```
Comment (niccokunzmann) [inline src/icalendar/tests/test_issue_867_todo_duration_fix.py]: Could you check out if this works here: `@pytest.mark.parametrize("component_class", [Event, Todo])`?

This way, we can avoid code duplication.

**Review by niccokunzmann — APPROVED** [review]
@SashankBhamidi Thanks for all your contributions! Would you like to resolve the merge conflict?

This looks good from my side, too and should fix the issue.

**Comment by niccokunzmann** [comment]
Thank you very much!
